Library/Sender.ts

```diff
@@ -128,59 +137,99 @@ class Sender {
     public saveOnCrash(payload: string) {
         this._storeToDiskSync(payload);
     }
-    
+
+    /**
+     * Send all temp files from last crash.
+     * reads all files that exist, deletes them and tries to send its load
+     */
+    public sendAllFilesOnDisk(): void {
+        var tempDir = Sender.OFFLINE_DIRECTORY || path.join(os.tmpDir(), Sender.TEMPDIR);
+
+        fs.exists(tempDir, (exists: boolean) => {
+            if (exists) {
+                fs.readdir(tempDir, (error, files) => {
+                    if (!error) {
+                        files = files.filter(f => path.basename(f).indexOf(".ai.json") > -1);
+                        if (files.length > 0) {
+                            files.forEach(file => {
+                                var filePath = path.join(tempDir, file);
+                                fs.readFile(filePath, (error, payload) => {
+                                    if (!error) {
+                                        // delete the file first to prevent double sending
```

Comment:
I would think that double telemetry would be better than losing any results. You may be using telemetry to alert on errors. I would rather have two errors then no errors reported.

However neither option is perfect. A new routine that is capable of cleaning cleaning up files only after a send is successful based on the contents of that file. Otherwise leave file on disk.